chisel/src/executor.rs

```diff
@@ -988,6 +998,31 @@ impl Type {
             .and_then(|ty| ty.try_as_ethabi(intermediate))
     }
 
+    fn get_function_return_type(
+        contract_expr: Option<&pt::Expression>,
+        intermediate: &IntermediateOutput,
+    ) -> Option<ParamType> {
+        let pt::Expression::FunctionCall(_, function_call, _) = contract_expr? else { return None };
+        let pt::Expression::MemberAccess(_, contract_name, function_name) = function_call.as_ref()
+        else {
+            return None
+        };
+        let pt::Expression::Variable(contract_name) = contract_name.as_ref() else { return None };
+        let pt::Expression::Variable(contract_name) =
+            intermediate.repl_contract_expressions.get(&contract_name.name)?
+        else {
+            return None
+        };
```

Comment:
I don't understand what's happening here 😅 can't we rewrite this so it's a little clearer and add docs?

Reply:
`contract_expr` represents the input of the user as a tree Expression. Im just trying to unwrap it and get to the string itself. I couldnt find an easier way to get to it except unwrapping each level 1 by 1.
If you have a better way, I'll change it.
I can add some comments if needed